### PR TITLE
chore: repair upgrade 18 proposal

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
## Description

Fix a variety of things in a3p-integration for Upgrade 18

 * The keys in updatePriceFeeds.js for specifying the platform don't match what's needed.
 * move test-localchain.js to z:acceptance
 * drop extraneous sdk-generate entries in package.json
 * drop eval.sh (and code it intended to run)
 * make upgrade.go and the parameterized proposals more robust to variations in the parameters
 * clean up go formatting

### Security Considerations

N/A

### Scaling Considerations

No impact

### Documentation Considerations
None

### Testing Considerations

We'll verify in DevNet

The logic for configurable core proposals was tested locally with the following `upgrade-next` config (in `package.json`):

```
  "agoricProposal": {
    "releaseNotes": false,
    "sdkImageTag": "unreleased",
    "planName": "UNRELEASED_BASIC",
    "upgradeInfo": {
      "coreProposals": {
        "steps": [
          [
            {
              "module": "@agoric/builders/scripts/inter-protocol/replace-electorate-core.js",
              "entrypoint": "defaultProposalBuilder",
              "args": [
                {
                  "config": {
                    "committeeName": "Economic Committee",
                    "voterAddresses": {
                      "gov1": "agoric1ee9hr0jyrxhy999y755mp862ljgycmwyp4pl7q",
                      "gov2": "agoric1wrfh296eu2z34p6pah7q04jjuyj3mxu9v98277",
                      "gov4": "agoric1c9gyu460lu70rtcdp95vummd6032psmpdx7wdy"
                    },
                    "highPrioritySendersConfig": {
                      "addressesToAdd": [
                        "agoric1c9gyu460lu70rtcdp95vummd6032psmpdx7wdy"
                      ],
                      "addressesToRemove": [
                        "agoric1ydzxwh6f893jvpaslmaz6l8j2ulup9a7x8qvvq"
                      ]
                    }
                  }
                }
              ]
            }
          ],
          [
            {
              "module": "@agoric/builders/scripts/inter-protocol/updatePriceFeeds.js",
              "entrypoint": "defaultProposalBuilder",
              "args": [
                {
                  "config": {
                    "oracleAddresses": [
                      "agoric1ee9hr0jyrxhy999y755mp862ljgycmwyp4pl7q",
                      "agoric1wrfh296eu2z34p6pah7q04jjuyj3mxu9v98277",
                      "agoric1ydzxwh6f893jvpaslmaz6l8j2ulup9a7x8qvvq"
                    ],
                    "inBrandNames": [
                      "ATOM",
                      "stATOM"
                    ]
                  }
                }
              ]
            }
          ],
          [
            "@agoric/builders/scripts/vats/add-auction.js"
          ],
          [
            "@agoric/builders/scripts/vats/upgradeVaults.js"
          ]
        ]
      }
    },
    "type": "Software Upgrade Proposal"
  },
```

### Upgrade Considerations

It's all upgrade.

### CODE REVIEW

**Review by mhofman — CHANGES_REQUESTED** [review]
This change looks fine, but I think there is another problem with the original. I believe that if the variant is `""`, the proposals are meant to be skipped altogether, but that doesn't seem to be what the logic is.

I believe currently it will perform a `process.exit()` and result in a crashloop of validator nodes.

**Comment by Chris-Hibbert** [comment]
>  I believe that if the variant is "", 

Where could that happen?

**Comment by mhofman** [comment]
> Where could that happen?

The basic or any non network specific upgrade name: https://github.com/Agoric/agoric-sdk/blob/ced5d490e498dd5426d2bf8164e156f0471ca30b/golang/cosmos/app/upgrade.go#L117-L120

**Comment by Chris-Hibbert** [comment]
I think this isn't complete. The variant name is also used in package.json

File: a3p-integration/proposals/n:upgrade-next/package.json [hunk 1]
```diff
@@ -16,10 +16,10 @@
       "inter-protocol/updatePriceFeeds.js price-feeds A3P_INTEGRATION",
       "vats/add-auction.js price-feeds",
       "vats/upgradeVaults.js price-feeds",
-      "inter-protocol/updatePriceFeeds.js submission/main main",
+      "inter-protocol/updatePriceFeeds.js submission/main MAINNET",
```
Comment (mhofman) [inline a3p-integration/proposals/n:upgrade-next/package.json]: Why is this even necessary?

Comment (Chris-Hibbert) [reply a3p-integration/proposals/n:upgrade-next/package.json]: It is used to generate the artifacts that can be uploaded (via cosgov, e.g.) to make a proposal. @dckc has been a strong advocate that those artifacts should be produced in CI.

Comment (mhofman) [reply a3p-integration/proposals/n:upgrade-next/package.json]: Um ok but we're not deploying this with a core eval afaik?

Is there a specific reason the upgrade-next layer needs to produce these artifacts ? and who is meant to use these artifacts?

Comment (dckc) [reply a3p-integration/proposals/n:upgrade-next/package.json]: true, now that they're part of a chain-halting upgrade, the artifacts are not really relevant

**Comment by Chris-Hibbert** [comment]
> I think this isn't complete. The variant name is also used in package.json

Fixed.

File: a3p-integration/proposals/n:upgrade-next/provisionPool.test.js
Comment (mhofman) [inline a3p-integration/proposals/n:upgrade-next/provisionPool.test.js:1]: Should this be in `z:acceptance` instead?

Edit: Nope, it needs to be split into `upgrade.go` and better tests.

Comment (Chris-Hibbert) [reply a3p-integration/proposals/n:upgrade-next/provisionPool.test.js:1]: Actually, my plan is to move it to upgrade 19. (it's a long story.)

**Comment by mhofman** [comment]
Tests for the config changes have passed locally!

**Review by mhofman — APPROVED** [review]
Got an LGTM from @michaelfig for the core proposal config changes assuming tests pass, which they have

**Pull request merged**